core/src/test/java/io/wcm/testing/mock/aem/dam/MockGraniteAssetWrapperTest.java

```diff
@@ -0,0 +1,131 @@
+package io.wcm.testing.mock.aem.dam;
+
+import com.adobe.granite.asset.api.Asset;
+import com.adobe.granite.asset.api.Rendition;
+import com.day.cq.dam.api.DamEvent;
+import io.wcm.testing.mock.aem.context.TestAemContext;
+import io.wcm.testing.mock.aem.junit.AemContext;
+import org.apache.commons.collections4.IteratorUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.testing.mock.sling.ResourceResolverType;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.osgi.service.event.EventHandler;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+
+public class MockGraniteAssetWrapperTest {
+
+    private static final byte[] BINARY_DATA = new byte[] {
+            0x01, 0x02, 0x03, 0x04, 0x05
+    };
+
+    @Rule
+    public final AemContext context = TestAemContext.newAemContext();
+
+    private Asset asset;
+    private MockAssetManagerTest.DamEventHandler damEventHandler;
```

Comment:
It would be nice to move `DamEventHandler` out of `MockAssetManagerTest`, so it can be reused without having to import `MockAssetManagerTest`. It would also be nice to provide the topics when registering the event handler, so that it will already only receive DAM events, the implementation can then be simplified to just expose the list of events (since all other methods are using that same list anyway) - A caller can then use `.stream()` to find the last event or the last 10, ...